code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/settings/SettingsFilter.java

```diff
@@ -19,6 +21,9 @@
     Set<String> publicSettings;
     boolean all;
     boolean canListAll;
+    List<String> secretSettings = Arrays.asList("api.auth.ldap.service.account.password", "api.auth.ldap.openldap.service.account.password",
+            "api.auth.azure.admin.password", "api.auth.github.client.secret");
+    HashSet<String> settingsHidden = new HashSet<>(secretSettings);
```

Comment:
Make this a constant (`private static final Set`)
Dont ever declare a variable as the underlying implmentation like HashSet. Always use the interface Set. This is true of all.
After declaring as a constant, you can use a static initializer to add the valus:
```
static {
  secretSettings.add("api...");
}
```
You should not have to create the List at all...just the set